### PR TITLE
Code review nits for semver grammar

### PR DESCRIPTION
**Pull request by hildjj**
See conversation in #300

### CODE REVIEW

File: examples/semver.peggy [hunk 1]
```diff
@@ -39,13 +33,17 @@ preReleaseIdentifier
...
+// is compared lexically.  Return a string.
 alphanumericIdentifier
-  = digit* nonDigit identifierChar*
+  = $(digit* nonDigit identifierChar*)
 
+// Any semantically significant numbers are turned into integers for
+// later numeric comparison.
 numericIdentifier
-  = '0' / (positiveDigit digit*)
+  = n:('0' / $(positiveDigit digit*)) { return parseInt(n, 10); }
```
Comment (polkovnikov-ph) [inline examples/semver.peggy:46]: Formally should parse be `BigInt(n)` instead of `parseInt(n, 10)`.

File: examples/semver.peggy [hunk 2]
```diff
@@ -57,4 +55,4 @@ digit
   = [0-9]
 
 positiveDigit
-  = [1-9]
\ No newline at end of file
+  = [1-9]
```
Comment (polkovnikov-ph) [inline examples/semver.peggy:58]: Any other places where `[1-9]` is used?

File: examples/semver.peggy (hunk 1 above)
Comment (hildjj) [reply examples/semver.peggy:46]: eh, I guess.  We don't have to worry about IE anymore.

File: examples/semver.peggy (hunk 2 above)
Comment (hildjj) [reply examples/semver.peggy:58]: I'd like to stay as close as possible to the original grammar.

**Comment by hildjj** [comment]
@dselman look good, make sense?

**Comment by MarcelBolten** [comment]
When I copy-paste the grammar into https://peggyjs.org/online.html (Firefox 101.0.1 or edge 102.0.1245.44) and add `0.0.4` as input I get:
```JSON
{
   "major": [ERROR],
   "minor": [ERROR],
   "patch": [ERROR],
   "pre": null,
   "build": null
}
```
Changing `BigInt()` back to `parseInt()` all works fine. Is it due to [jsDump](https://github.com/peggyjs/peggy/blob/d603d740785e47c2fc5488fc6ac15dc6bb91a095/docs/js/online.js#L168)?

**Comment by hildjj** [comment]
> Is it due to [jsDump](https://github.com/peggyjs/peggy/blob/d603d740785e47c2fc5488fc6ac15dc6bb91a095/docs/js/online.js#L168)?

Probably.  We could switch to [node-inspect-extracted](https://github.com/hildjj/node-inspect-extracted), at least I know that works.

**Pull request merged**